### PR TITLE
Implemented: Support to fetch brands from backend on settings page#2r69y7g

### PR DESCRIPTION
**Pull request by shashwatbangar**
## This PR is dependent on - https://github.com/hotwax/preorder/pull/45

Closes #

### Short Description and Why It's Useful

Earlier we were fetching BRANDS from environment file, now we are fetching from backend
As to ionic components work as expected, this app needed an upgradation to ionic 6, so this PR is dependent on mentioned PR as upgradation work is done in that 

### Screenshots of Visual Changes before/after (If There Are Any)

**IMPORTANT NOTICE** - Remember to update `CHANGELOG.md` with description of your change

### Contribution and Currently Important Rules Acceptance

- [ ] I read and followed [contribution rules](https://github.com/DivanteLtd/vsf-capybara/blob/master/CONTRIBUTING.md)

### CODE REVIEW

File: src/store/modules/user/actions.ts [hunk 1]
```diff
@@ -125,6 +154,17 @@ const actions: ActionTree<UserState, RootState> = {
     */
      setUserInstanceUrl ({ state, commit }, payload){
       commit(types.USER_INSTANCE_URL_UPDATED, payload)
-    }
+    },
+
+  /**
+   * Set Ecom store
+   */
+    async setEcomStore({ commit, dispatch }, payload) {
```
Comment (adityasharma7) [inline src/store/modules/user/actions.ts:162]: I think already have an action `setSelectedBrand` for the same. Same for getters. I think we could use the EcomStore for consistency across the apps, we could use Brand label on settings page, but use eCom store across the app

**Review by adityasharma7 — CHANGES_REQUESTED** [review]

File: src/store/modules/user/actions.ts (hunk 1 above)
Comment (shashwatbangar) [reply src/store/modules/user/actions.ts:148]: Got it sir. Fixed it

File: src/views/settings.vue [hunk 2]
```diff
@@ -103,9 +102,13 @@ export default defineComponent({
       });
       return timeZoneModal.present();
     },
-    updateBrand(event: any) {
-      this.store.dispatch("user/setSelectedBrand", { selectedBrand: event.detail.value})
-    }
+    updateBrand(store: any) {
+      if(this.userProfile) {
+        this.store.dispatch('user/setEcomStore', {
+          'eComStore': this.userProfile.stores.find((str: any) => str.productStoreId == store['detail'].value)
```
Comment (adityasharma7) [inline src/views/settings.vue:108]:
```suggestion
          'eComStore': this.userProfile.stores.find((store: any) => str.productStoreId == store['detail'].value)
```

File: src/views/settings.vue [hunk 3]
```diff
@@ -103,9 +102,13 @@ export default defineComponent({
       });
       return timeZoneModal.present();
     },
-    updateBrand(event: any) {
-      this.store.dispatch("user/setSelectedBrand", { selectedBrand: event.detail.value})
-    }
+    updateBrand(store: any) {
```
Comment (adityasharma7) [inline src/views/settings.vue:105]:
```suggestion
    updateBrand(event: any) {
```

File: src/store/modules/user/index.ts [hunk 4]
```diff
@@ -10,8 +10,11 @@ const userModule: Module<UserState, RootState> = {
     state: {
       token: '',
       current: null,
-      selectedBrand: '',
       instanceUrl: '',
+      currentEComStore: {
+        productStoreId: "",
```
Comment (adityasharma7) [inline src/store/modules/user/index.ts:15]: Keep it empty, we will not support None here

File: src/store/modules/user/actions.ts [hunk 5]
```diff
@@ -80,15 +80,44 @@ const actions: ActionTree<UserState, RootState> = {
...
+        "noConditionFind": "Y"
+      }
+
+      const storeResp = await UserService.getEComStores(payload);
+      if(storeResp.status === 200 && !hasError(storeResp) && storeResp.data.docs?.length > 0) {
+        const stores = storeResp.data.docs;
+
+        resp.data.stores = [
+          ...(stores ? stores : []),
+          {
```
Comment (adityasharma7) [inline src/store/modules/user/actions.ts:102]: This is not needed, we will not support None in preorder app

**Review by adityasharma7 — CHANGES_REQUESTED** [review]

File: src/store/modules/user/actions.ts (hunk 5 above)
Comment (shashwatbangar) [reply src/store/modules/user/actions.ts:102]: Done sir

File: src/store/modules/user/index.ts (hunk 4 above)
Comment (shashwatbangar) [reply src/store/modules/user/index.ts:15]: ok sir

File: src/views/settings.vue (hunk 3 above)
Comment (shashwatbangar) [reply src/views/settings.vue]: done sir

File: src/views/settings.vue (hunk 2 above)
Comment (shashwatbangar) [reply src/views/settings.vue]: ok sir

**Review by adityasharma7 — CHANGES_REQUESTED** [review]

File: src/store/modules/user/actions.ts [hunk 6]
```diff
@@ -107,24 +136,24 @@ const actions: ActionTree<UserState, RootState> = {
...
       this.dispatch("order/resetOrderQuery")
       await UserService.setUserPreference({
         'userPrefTypeId': 'SELECTED_BRAND',
-        'userPrefValue': payload.selectedBrand
+        'userPrefValue': payload.eComStore.productStoreId
       });
     },
-
-  /**
-    * Set User Instance Url
```
[the remark this answers is not in the document]
Comment (shashwatbangar) [reply src/store/modules/user/actions.ts:124]: reverted it sir.

File: src/store/modules/user/actions.ts [hunk 7]
```diff
@@ -80,15 +80,44 @@ const actions: ActionTree<UserState, RootState> = {
...
+      const stores = resp.data.stores
+      let userPrefStore = ''
+
+      try {
+        const userPref =  await UserService.getUserPreference({
+          'userPrefTypeId': 'SELECTED_BRAND'
+        });
+        userPrefStore = stores.find((store: any) => store.productStoreId == userPref.data.userPrefValue)
+      } catch (err) {
+        console.log(err)
```
[the remark this answers is not in the document]
Comment (shashwatbangar) [reply src/store/modules/user/actions.ts:118]: done sir

File: src/store/modules/user/actions.ts (hunk 5 above)
Comment (shashwatbangar) [reply src/store/modules/user/actions.ts]: Done sir

File: src/store/modules/user/actions.ts [hunk 8]
```diff
@@ -80,19 +80,49 @@ const actions: ActionTree<UserState, RootState> = {
...
+      }
+      dispatch('setEcomStore', { eComStore: userPrefStore ? userPrefStore : stores.length > 0 ? stores[0] : {} });
       commit(types.USER_INFO_UPDATED, resp.data);
     }
   },
 
+  /**
+   * Set User Instance Url
+   */
+  setUserInstanceUrl ({ state, commit }, payload){
```
[the remark this answers is not in the document]
Comment (shashwatbangar) [reply src/store/modules/user/actions.ts:122]: done sir

File: src/store/modules/user/actions.ts [hunk 9]
```diff
@@ -80,19 +80,49 @@ const actions: ActionTree<UserState, RootState> = {
...
+        "entityName": "ProductStore",
+        "distinct": "Y",
+        "noConditionFind": "Y"
+      }
+
+      const storeResp = await UserService.getEComStores(payload);
+      let stores = storeResp.data.docs
+      if(storeResp.status === 200 && !hasError(storeResp) && storeResp.data.docs?.length > 0) {
+        stores = [
+          ...(stores ? stores : [])
```
[the remark this answers is not in the document]
Comment (shashwatbangar) [reply src/store/modules/user/actions.ts:100]: got it sir. fixed it

File: src/store/modules/user/actions.ts [hunk 10]
```diff
@@ -80,19 +80,49 @@ const actions: ActionTree<UserState, RootState> = {
...
+          "storeName_op": "not-empty"
+        },
+        "fieldList": ["productStoreId", "storeName"],
+        "entityName": "ProductStore",
+        "distinct": "Y",
+        "noConditionFind": "Y"
+      }
+
+      const storeResp = await UserService.getEComStores(payload);
+      let stores = storeResp.data.docs
```
[the remark this answers is not in the document]
Comment (shashwatbangar) [reply src/store/modules/user/actions.ts:97]: ok sir

**Pull request merged**